domain/src/main/java/org/oppia/domain/oppialogger/exceptions/OppiaUncaughtExceptionHandler.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.domain.oppialogger.exceptions
+
+import org.oppia.util.system.OppiaClock
+import javax.inject.Inject
+
+/** Handler for catching fatal exceptions before the defaultUncaughtExceptionHandler. */
+class OppiaUncaughtExceptionHandler @Inject constructor(
+  private var exceptionsController: ExceptionsController,
+  private var oppiaClock: OppiaClock
+) : Thread.UncaughtExceptionHandler {
+
+  /** Logs an uncaught exception to the [exceptionsController]. */
+  override fun uncaughtException(thread: Thread?, throwable: Throwable?) {
```

Comment:
Dont we need to throw an exception here too? Since the caller is catching it?

Reply:
No, I don't think so. This function is called whenever the exception handler is invoked so I think the throwable will go forward.

Reply:
Can we do the try {} catch logic in the OppiaApplication here instead? That way if we want to register the handler anywhere else it would have all the try catch logic set up.

Reply:
Yes, done.